.github/workflows/docs.yml

```diff
@@ -0,0 +1,55 @@
+name: Documentation CI
+on:
+  pull_request:
+    paths:
+      - .vale.ini
+      - docs/**/*.md
+      - Library/Homebrew/**/*.rb
+
+permissions:
+  contents: read
+
+jobs:
+  docs:
+    if: github.repository == 'Homebrew/brew'
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        working-directory: docs
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Run vale for docs linting
+        run: |
+          brew install vale
+          vale .
+
+      - name: Install Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "2.7"
+          bundler-cache: true
+          working-directory: docs
+
+      - name: Check Markdown syntax
+        run: |
+          bundle exec rake lint || :
+
+      - name: Build docs site
+        run: bundle exec rake build
+
+      - name: Checkout Homebrew/rubydoc.brew.sh
+        uses: actions/checkout@main
+        with:
+          repository: Homebrew/rubydoc.brew.sh
+          path: rubydoc
+
+      - name: Process rubydoc comments
+        working-directory: Library/Homebrew
+        env:
+          BUNDLE_GEMFILE: ../../rubydoc/Gemfile
+        run: |
+          bundle install --jobs 4 --retry 3
+          bundle exec yard doc --plugin sorbet --no-output #--fail-on-warning
```

Comment:
I reckon it's maybe work splitting this into a separate job given it's working on a different Bundler? Thoughts?

Reply:
@EricFromCanada Sorry, I mean the two differing `Gemfile`/`BUNDLE_GEMFILE` bits here may make sense to be in different GitHub Actions jobs in the same file. Thoughts?

Reply:
They're now in separate jobs. Only outstanding item is the Test Everything check which seems to be unrelated.